.github/workflows/build-publish.yml

```diff
@@ -0,0 +1,199 @@
+        tenant-id: ${{ vars.AZURE_TENANT_ID }}
+        subscription-id: ${{ vars.AZURE_SUBSCRIPTION_ID }}
+    - name: Authenticate to ACR
+      run: az acr login -n ${{ vars.AZURE_REGISTRY_SERVER }}
+    # Perform a 'docker login' so that we can push to the current repo's GHCR. We will push the unpatched
+    # images to GHCR first, run a Trivy scan and Copa patch on those, and then push the patched images
+    # to both GHCR and ACR.
+    - name: Authenticate to GHCR
+      uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    # Work out the target image references (registry images and tags) for publishing.
+    - name: Get Image References
+      id: getref
+      run: |
+        $tag_id = "${{ needs.common.outputs.tag_id_base }}-${{ matrix.tagext }}"
+        echo "tag_id=$tag_id" >> $env:GITHUB_OUTPUT
+        echo "acr_image_ref=${{ needs.common.outputs.acr_image_id }}:$tag_id" >> $env:GITHUB_OUTPUT
+        echo "ghcr_image_ref=${{ needs.common.outputs.ghcr_image_id }}:$tag_id" >> $env:GITHUB_OUTPUT
+        echo "${{ matrix.os }}-acr-image-ref=${{ needs.common.outputs.acr_image_id }}:$tag_id" >> $env:GITHUB_OUTPUT
+        echo "${{ matrix.os }}-ghcr-image-ref=${{ needs.common.outputs.ghcr_image_id }}:$tag_id" >> $env:GITHUB_OUTPUT
+    - name: Build Image
+      id: build
+      run: docker build -f ${{ matrix.file }} --build-arg BASE_IMAGE=${{ matrix.baseimage }} -t ${{ steps.getref.outputs.ghcr_image_ref }} .
+    # If this OS supports patching images using Copa, push the unpatched image to GHCR with the tag suffix '-unpatched'.
+    # Then, scan and patch this image, and re-tag it with the original tag (no suffix).
+    - name: Push Unpatched Image to GHCR
+      if: matrix.canpatch
+      run: |
+        docker tag ${{ steps.getref.outputs.ghcr_image_ref }} ${{ steps.getref.outputs.ghcr_image_ref }}-unpatched
+        docker push ${{ steps.getref.outputs.ghcr_image_ref }}-unpatched
+    - name: Generate Trivy Report
+      if: matrix.canpatch
+      uses: aquasecurity/trivy-action@41f05d9ecffa2ed3f1580af306000f734b733e54 # v0.11.2
+      with:
+        scan-type: 'image'
+        format: 'json'
+        output: 'report.json'
+        ignore-unfixed: true
+        vuln-type: 'os'
+        image-ref: ${{ steps.getref.outputs.ghcr_image_ref }}-unpatched
+    - name: Check Vuln Count
+      if: matrix.canpatch
+      id: vulncount
+      run: |
+        $report_file = "report.json"
+        cat $report_file
+        $vuln_count = jq '.Results[0].Vulnerabilities | length' $report_file
+        echo "vuln_count=$vuln_count" >> $env:GITHUB_OUTPUT
+    - name: Copa Action
+      if: matrix.canpatch && steps.vulncount.outputs.vuln_count != '0'
+      id: copa
+      uses: project-copacetic/copa-action@1eb86b0907bce48225b66dc9488c7d329c2d48a0 # v1.0.0
```

Comment:
Note: need to upgrade to 1.0.1 which adds a curl retry when getting the latest copa version.

Reply:
Updated, but now waiting for the next release of copa which will include https://github.com/project-copacetic/copacetic/pull/377